contracts/SnapshotToken/Helpers/TokenMetadata.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity 0.4.15;
+
+import '../../Standards/ITokenMetadata.sol';
+
+contract TokenMetadata is ITokenMetadata {
+
+    string public tokenName;                //The Token's name: e.g. DigixDAO Tokens
+    uint8 public tokenDecimals;             //Number of decimals of the smallest unit
+    string public tokenSymbol;              //An identifier: e.g. REP
+    string public tokenVersion;             //An arbitrary versioning scheme
+
+    /// @notice Constructor to create a MiniMeToken
+    /// @param _tokenName Name of the new token
+    /// @param _decimalUnits Number of decimals of the new token
+    /// @param _tokenSymbol Token Symbol for the new token
+    function TokenMetadata(
+        string _tokenName,
+        uint8 _decimalUnits,
+        string _tokenSymbol,
+        string _version
+    ) {
+        tokenName = _tokenName;                                 // Set the name
+        tokenDecimals = _decimalUnits;                          // Set the decimals
+        tokenSymbol = _tokenSymbol;                             // Set the symbol
+        tokenVersion = _version;
+    }
+
+    function symbol()
```

Comment:
This contract has two sets of functions returning same value. There's `symbol()` and there's `tokenSymbol()` which compiler generates. Why so?

Reply:
Ok got it, you want interface. If you use `string public token;` it will not auto-implement by compiler ;>?
Should at least `string public tokenName; ` be internal to not expose same data twice?

Reply:
> If you use string public token; it will not auto-implement by compiler ;>?

It will, but Solidity does not count it as implementing the interface, so it will think the contract is abstract and not produce code, but also not fail ( == stupid).

You are right, I should make it  private.